OneCommunityGlobal/HGNRest
# Jerry - email verification admin accounts

# Description
<img width="700" alt="Screenshot 2023-09-28 at 8 53 11 AM" src="https://github.com/OneCommunityGlobal/HGNRest/assets/55968519/b65d1208-b767-436a-8fda-565aba0a57d7">

## Related PRS (if any):
- Front End PR #1446 (https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1446)

## Main changes explained:
- added `actualEmail` field to userProfile schema
- check for Owner and Administrator roles before making request to Beta Login route
- minor formatting fix in 'Acutal Password' label
- added axios to package.lock (I was unable to use fetch api successfully)


## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Follow the instructions of the front end PR #1446 (https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1446)


## Note:
~~If anyone knows how to use Fetch API instead of Axios to successfully make the request to the Beta Login route on line 145 of `src/controllers/userProfileController.js`, then please do let me know. This is because we can avoid adding the additional axios library if we can use Fetch instead. Thanks!~~

In my latest commit, I replaced the axios call with a fetch call.

Before (with axios):
```
  try {
  // Log in to Beta login route using provided credentials
  let response = await axios.post(url + "login", {
    email: email,
    password: password
  })
} catch (error) {
  res.status(400).send({
    error: 'The actual email or password you provided is incorrect. Please enter the actual email and password associated with your account in the Main HGN app.',
    type: 'credentials',
  });
  return;
}
```

After (with fetch):
```
try {
  // Log in to Beta login route using provided credentials
    const response = await fetch(url + 'login', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({ email, password }),
  });
  if (!response.ok) {
    throw new Error('Invalid credentials');
  }
} catch (error) {
  res.status(400).send({
    error: 'The actual email or password you provided is incorrect. Please enter the actual email and password associated with your account in the Main HGN app.',
    type: 'credentials',
  });
  return;
}
```
